src/app/features/admin/components/route-form/route-form/route-form.component.ts

```diff
@@ -0,0 +1,236 @@
+import { NgForOf } from '@angular/common';
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  computed,
+  DestroyRef,
+  effect,
+  inject,
+  input,
+  output,
+  signal,
+} from '@angular/core';
+import { takeUntilDestroyed } from '@angular/core/rxjs-interop';
+import { NonNullableFormBuilder, ReactiveFormsModule, Validators } from '@angular/forms';
+import { AlertService } from '@core/services/alert/alert.service';
+import { RouteCardComponent } from '@features/admin/components/route-card/route-card.component';
+import { ConnectedStation, Station } from '@features/admin/interfaces/station.interface';
+import { TrainRoute } from '@features/admin/interfaces/train-route.interface';
+import { RouteApiService } from '@features/admin/services/route-api/route-api.service';
+import { MIN_ROUTE_FORM_CONTROL_COUNT } from '@shared/constants/min-route-control-count';
+import { TuiButton, TuiDataList, TuiIcon, TuiLoader, TuiSurface, TuiTitle } from '@taiga-ui/core';
+import { TuiAvatar } from '@taiga-ui/kit';
+import { TuiCardLarge, TuiHeader } from '@taiga-ui/layout';
+import { TuiSelectModule, TuiTextfieldControllerModule } from '@taiga-ui/legacy';
+import { map } from 'rxjs';
+
+enum ControlsType {
+  path = 'path',
+  carriages = 'carriages',
+}
+
+@Component({
+  selector: 'dd-route-form',
+  standalone: true,
+  imports: [
+    RouteCardComponent,
+    TuiCardLarge,
+    TuiSurface,
+    TuiSelectModule,
+    TuiTextfieldControllerModule,
+    ReactiveFormsModule,
+    NgForOf,
+    TuiDataList,
+    TuiAvatar,
+    TuiButton,
+    TuiIcon,
+    TuiHeader,
+    TuiTitle,
+    TuiLoader,
+  ],
+  templateUrl: './route-form.component.html',
+  styleUrl: './route-form.component.scss',
+  changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class RouteFormComponent {
+  public readonly trainRoute = input<TrainRoute>();
+
+  public save = output();
+
+  private readonly alert = inject(AlertService);
+
+  private readonly routeApiService = inject(RouteApiService);
+
+  private readonly destroy = inject(DestroyRef);
+
+  private readonly fb = inject(NonNullableFormBuilder);
+
+  private readonly cdr = inject(ChangeDetectorRef);
+
+  protected readonly connectedToStations = signal<string[][]>([]);
+
+  protected readonly stations = signal<Station[]>([]);
+
+  protected readonly carriages = signal<string[]>([]);
+
+  protected readonly ControlsType = ControlsType;
+
+  private readonly allAvailableStations = computed(() =>
+    this.stations().map((station) => {
+      return {
+        id: station.id,
+      };
```

Comment:
because it is equivalent to the `connectedTo` array at each station, which contains all stations.
`connectedTo` array contains such objects
```ts
{id: 54, distance: 845}
```